admin/server/auth/pkce.go

```diff
@@ -0,0 +1,178 @@
+package auth
+
+import (
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/rilldata/rill/cli/pkg/auth"
```

Comment:
nit: we currently only have imports from cli -> admin, not from admin -> cli. Can we move this to `admin/pkg` instead?